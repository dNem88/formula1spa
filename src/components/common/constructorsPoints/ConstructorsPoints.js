import React from 'react'
import styles from './constructorsPoints.module.css'


function ConstructorsPoints(props) {
    
    return (
        <div className={styles['driver-row']}>
            <p>{props.position}</p>
            <p>{props.constructor}</p>
            <p></p>
            <p>{props.nationality}</p>
            <p>{props.points}</p>
        </div>
    )
}

export default ConstructorsPoints