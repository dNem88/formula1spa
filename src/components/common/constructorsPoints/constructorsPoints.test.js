import React from 'react'
import renderer from 'react-test-renderer'
import ConstructorsPoints from './ConstructorsPoints'
import {MemoryRouter} from 'react-router-dom'
import {screen, render} from '@testing-library/react'

const mockProps = {
    position: '7',
    constructor: 'Mercedes',
    nationality: 'German',
    points: '100'
}
describe('Testing Constructor Points Component', () => {
    it('render properly based on props', () => {
        const snapshot = renderer.create(
            <MemoryRouter>
                <ConstructorsPoints {...mockProps}/>
            </MemoryRouter>
        ).toJSON()
        expect(snapshot).toMatchSnapshot()
        expect(snapshot.props.className).toBe('driver-row')
        expect(snapshot.type).toBe('div')
    })
    test('p-tags have correct values', () => {
        render(<ConstructorsPoints {...mockProps}/>)
        expect(screen.getByText(mockProps.points)).toBeTruthy()
        expect(screen.getByText(mockProps.nationality)).toBeTruthy()
        expect(screen.getByText(mockProps.constructor)).toBeTruthy()
        expect(screen.getByText(mockProps.position)).toBeTruthy()
        
    })
})