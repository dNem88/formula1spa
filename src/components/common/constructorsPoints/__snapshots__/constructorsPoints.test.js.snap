// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Constructor Points Component render properly based on props 1`] = `
<div
  className="driver-row"
>
  <p>
    7
  </p>
  <p>
    Mercedes
  </p>
  <p />
  <p>
    German
  </p>
  <p>
    100
  </p>
</div>
`;
